src/float.rkt

```diff
@@ -80,29 +80,34 @@
    [(and (real? x1) (real? x2))
     (cond [(nan? x1) #f] [(nan? x2) #t] [else (< x1 x2)])]
    [else
+    (define x1* (real->repr x1 repr))
+    (define x2* (real->repr x2 repr))
     (cond
-     [(special-value? x1 repr) #f]
-     [(special-value? x2 repr) #t]
-     [else (< ((representation-repr->ordinal repr) x1)
-              ((representation-repr->ordinal repr) x2))])]))
+     [(special-value? x1* repr) #f]
+     [(special-value? x2* repr) #t]
+     [else (< ((representation-repr->ordinal repr) x1*)
+              ((representation-repr->ordinal repr) x2*))])]))
 
 (define (<=/total x1 x2 repr)
   (or (</total x1 x2 repr) (=-or-nan? x1 x2 repr)))
 
 (define (value->json x repr)
-  (match x
+  (define x* (real->repr x repr))
+  (match x*
```

Comment:
Kind of ugly. Why not match directly on the `real→repr` result and bind that in the branches where it's used?